File: 1813-sentence-similarity-iii/1813-sentence-similarity-iii.cpp
class Solution {
public:
    vector<string> split(string &str) {
        vector<string>ans;
        str += ' ';
        string s = "";
        for(char c:str) {
            if(c == ' ') {
                ans.push_back(s);
                s = "";
            }
            else {
                s += c;
            }
        }
        return ans;
    }
    bool areSentencesSimilar(string sentence1, string sentence2) {
        vector<string>s1 = split(sentence1);
        vector<string>s2 = split(sentence2);
        if(s1.size() < s2.size()) swap(s1, s2);
        int start = 0, end = 0;
        while(start < s2.size() && s1[start] == s2[start]) start++;
        while(end < s2.size() && s2[s2.size() - 1 - end] == s1[s1.size() - 1 - end]) end++;
        return start + end >= s2.size();
    }
};

File: 1813-sentence-similarity-iii/1813-sentence-similarity-iii.java
class Solution {
    public boolean areSentencesSimilar(String sentence1, String sentence2) {
        String[] s1 = sentence1.split(" ");
        String[] s2 = sentence2.split(" ");
        if(s1.length < s2.length) {
            String[] temp = s1;
            s1 = s2;
            s2 = temp;
        }
        int start = 0, end = 0;
        while(start < s2.length && s1[start].equals(s2[start])) start++;
        while(end < s2.length && s1[s1.length - 1 - end].equals(s2[s2.length - 1 - end])) end++;
        return start + end >= s2.length;
    }
}